opentracing-api/src/main/java/io/opentracing/BaseSpan.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+import java.util.Map;
+
+/**
+ * {@link BaseSpan} represents the OpenTracing specification's Span contract with the exception of methods to finish
+ * said Span. For those, either use {@link Span#finish()} or {@link ActiveSpan#deactivate()} depending on the
+ * programming model.
+ *
+ * @see Span
+ * @see ActiveSpan
+ * @see Tracer.SpanBuilder#startManual()
+ * @see Tracer.SpanBuilder#startActive()
+ */
+public interface BaseSpan {
+    /**
+     * Retrieve the associated SpanContext.
+     *
+     * This may be called at any time, including after calls to finish().
+     *
+     * @return the SpanContext that encapsulates Span state that should propagate across process boundaries.
+     */
+    SpanContext context();
+
+    /**
+     * Set a key:value tag on the Span.
+     */
+    // overloaded 3x to support the BasicType concern
+    BaseSpan setTag(String key, String value);
```

Comment:
@cwe1ss this is true, but not really a new problem... `ActiveSpan` methods used to return a `Span` which also has no `capture()` method. This is not wonderful but *is* livable, at least in my mind... the chaining is a convenience that doesn't need to be part of the same expression as finish()ing or capture()ing.

We could alternatively override everything and specify tighter types on the chaining methods, but IMO it's not worth it.